x-pack/test/security_solution_endpoint/apps/endpoint/artifact_entries_list.ts

```diff
@@ -0,0 +1,517 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { unzip } from 'zlib';
+import { promisify } from 'util';
+import expect from '@kbn/expect';
+import { TranslatedExceptionListItem } from '@kbn/security-solution-plugin/server/endpoint/schemas/artifacts/lists';
+import { ArtifactElasticsearchProperties } from '@kbn/fleet-plugin/server/services/artifacts/types';
+import { IndexedHostsAndAlertsResponse } from '@kbn/security-solution-plugin/common/endpoint/index_data';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  const pageObjects = getPageObjects(['common', 'blocklist', 'hostIsolationExceptions']);
+  const testSubjects = getService('testSubjects');
+  const browser = getService('browser');
+  const endpointTestResources = getService('endpointTestResources');
+  const policyTestResources = getService('policyTestResources');
+  const retry = getService('retry');
+  const esClient = getService('es');
+  const unzipPromisify = promisify(unzip);
+
+  describe('For each artifact list under management', function () {
+    let indexedData: IndexedHostsAndAlertsResponse;
+
+    interface ArtifactResponseType {
+      _index: string;
+      _id: string;
+      _score: number;
+      _source: ArtifactElasticsearchProperties;
+    }
+
+    interface ArtifactBodyType {
+      entries: TranslatedExceptionListItem[];
+    }
+
+    const testsData = [
```

Comment:
could we make these data items functions that return the data? just trying to avoid potential for it to be mutated and have an impact on all tests that use it. (same for the data below)

Reply:
Also,
Maybe to keep the test file focused only on the test, maybe move the test data into a separate file (`mocks.ts`?) and import it from there? 😄

Reply:
Yeah, agree on both. Will have to define a function param if we move it outside the test to get the `pageObjects`, but I think it make sense